action/protocol/generic_validator.go

```diff
@@ -85,6 +85,15 @@ func (v *GenericValidator) Validate(ctx context.Context, selp *action.SealedEnve
 				return action.ErrNonceTooLow
 			}
 		}
+		if ok && !featureCtx.EnableAccessListTx && selp.Version() == action.AccessListTxType {
+			return errors.Wrap(action.ErrInvalidAct, "access list tx is not enabled")
+		}
+		if ok && !featureCtx.EnableDynamicFeeTx && selp.Version() == action.DynamicFeeTxType {
+			return errors.Wrap(action.ErrInvalidAct, "dynamic fee tx is not enabled")
+		}
+		if ok && !featureCtx.EnableBlobTransaction && selp.Version() == action.BlobTxType {
+			return errors.Wrap(action.ErrInvalidAct, "blob tx is not enabled")
+		}
```

Comment:
should also do this check in `workingset.go`?

Reply:
I think it's enough to limit it only in actpool and block validation. For honest nodes, mintblock will not have these transactions due to the restrictions in actpool. For malicious nodes, even if they include these transactions in a block, other nodes will discard the block during validation.